packages/core/src/fhir-to-cda/cda-types/shared-types.ts

```diff
@@ -73,37 +97,47 @@ export type CdaName = {
   validTime: CdaPeriod;
 };
 
+export type CDAOriginalText = {
+  reference: {
+    [_valueAttribute]: string;
+  };
+};
+
 // Ce (CE) stands for Coded with Equivalents
 export type CdaCodeCe = {
-  _codeAttribute?: string;
-  _codeSystemAttribute?: string;
-  _codeSystemNameAttribute?: string;
-  _displayNameAttribute?: string;
+  [_codeAttribute]?: string;
+  [_codeSystemAttribute]?: string;
+  [_codeSystemNameAttribute]?: string;
+  [_displayNameAttribute]?: string;
```

Comment:
That's the only way to make it use the constants for these attributes. Without the brackets, it'll literally set `object._codeAttribute`, rather than `object._code`

Reply:
I don't think I fully understand what you mean. It would be useful to share a JSFiddle w/ the issue so we can understand what's going on - and try to help/find a solution. Defs not good to have square brackets when accessing an object property, from what I can see.

Reply:
I think [this example](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBA+mAhgWwKYEEpQE4EsBGArlKjALwwBECKqlA3AFCNQCeADqQMIAmiAKh1IUA3owCQNNJhwFiqAFwxoeMAHNGAX2ahIsbClwAbAHK0lvAUPIwxkpNKx4iJJZQBKho5S07wEECNUADojEDUACgNkYzM0AEomXQCg0PCorziQqQwnORJE5jZOGC4QbGxUYChLQRLRRhgmmABtHJlneQBdJRVcdV9GZP1EADMsgDFEYwtyyura6wbm+Adc2RdFKk9Rn20h-xHx2gBlQmBgVAgIWYqqmr464VtG5ra1jvzUHu2xvYPIIEQmFIgZjmgpsZCskgWlQWMsmcLlcIND-LCQRlwagkZdrsF2nlNvEASlgekwYjzniIO9aJ9Nl0STDUpjKadqSiCWt4kA) shows what I mean

Reply:
Oh, I see what you mean. We're creating types w/ the property names being defined elsewhere.

I'm wondering if there are cases where it makes sense, maybe for attribute names that can't be represented in TS?

Because in this case, I'd much rather use

```ts
    return {
      component: {
        observation: {
          _classCode: "OBS",
          _moodCode: "EVN",

```

...than

```ts
    return {
      component: {
        observation: {
          [_classCodeAttribute]: "OBS",
          [_moodCodeAttribute]: "EVN",
```

Reply:
Will remove the attribute const for most attributes. Will only keep the complicated ones, i.e. `_xsi:type`

Reply:
Done